File: src/leetcode/python/FindMinimumInRotatedSortedArray_153.py
class FindMinimumInRotatedSortedArray_153(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, j = 0, len(nums)-1
        while i < j:
            m = (i + j) / 2
            if nums[m] > nums[j]:
                i = m + 1
            else:
                j = m
        return nums[i]

File: src/leetcode/java/FindMinimumInRotatedSortedArray_153.java
package leetcode.java;

public class FindMinimumInRotatedSortedArray_153 {
  public int findMin(int[] nums) {
    return min(nums, 0, nums.length - 1);
  }

  private static int min(int[] a, int begin, int end) {
    if (begin > end) return a[0];
    int mid = (end + begin) / 2;
    if (mid > 0 && a[mid - 1] > a[mid]) return a[mid];
    if (mid < a.length - 1 && a[mid] > a[mid + 1]) return a[mid + 1];
    if (a[mid] < a[end]) return min(a, begin, mid - 1);
    return min(a, mid + 1, end);
  }
}
